src/parser/scanner.lex

```diff
@@ -249,20 +249,64 @@ IP_OCTET                    ([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])
                                 return TokenType::IPV4;
                             }
 0[Xx]{HEX}+                 {
+                                if (yyleng > 18) {
+                                    auto i = 2;
+                                    while (i < yyleng && yytext[i] == '0') {
+                                        i++;
+                                    }
+                                    if (yyleng - i > 16) {
+                                        yyterminate();
+                                    }
+                                }
                                 int64_t val = 0;
                                 sscanf(yytext, "%lx", &val);
```

Comment:
Should we judge the return value of sscanf?

Reply:
Yes, we should have done that, even though the regex and checking above already guarantee the correctness.

Reply:
👍